java/configuration/src/main/java/sleeper/configuration/properties/table/TableProperty.java

```diff
@@ -303,6 +305,16 @@ public interface TableProperty extends SleeperProperty {
             .description("The number of days that transaction log snapshots remain in the snapshot store before being deleted.")
             .propertyGroup(TablePropertyGroup.METADATA)
             .build();
+    TableProperty TRANSACTION_LOG_NUMBER_BEHIND_TO_DELETE = Index.propertyBuilder("sleeper.table.metadata.transactionlog.delete.number.behind.latest.snapshot")
+            .defaultProperty(DEFAULT_TRANSACTION_LOG_NUMBER_BEHIND_TO_DELETE)
+            .description("The minimum number of transactions that a transaction must be behind the latest snapshot before being deleted.")
+            .propertyGroup(TablePropertyGroup.METADATA)
+            .build();
+    TableProperty TRANSACTION_LOG_MINUTES_BEHIND_TO_DELETE = Index.propertyBuilder("sleeper.table.metadata.transactionlog.delete.mins.behind.latest.snapshot")
+            .defaultProperty(DEFAULT_TRANSACTION_LOG_MINUTES_BEHIND_TO_DELETE)
+            .description("The minimum number of minutes that a transaction must exist for before the latest snapshot in order to be deleted.")
+            .propertyGroup(TablePropertyGroup.METADATA)
+            .build();
```

Comment:
We can give a bit more information in these descriptions about what to be aware of, and which other properties to compare these settings against.